bodo/io/arrow.cpp

```diff
@@ -47,8 +47,8 @@ ArrowReader* pq_reader_init_py_entry(
 
 // --------- functions defined in iceberg_parquet_reader.cpp --------
 table_info* iceberg_pq_read_py_entry(
-    const char* conn, const char* database_schema, const char* table_name,
-    bool parallel, int64_t tot_rows_to_read, PyObject* iceberg_filter_str,
+    const char* conn, const char* table_id, bool parallel,
```

Comment:
I made a small tweak in various parts of the code to use the Iceberg table_id concept, which is basically a `.` delimited list of namespaces (and the last is the table name). The old approach with a separate `database_schema` and `table_name` can be awkward because we need to set `database_schema = ""` for certain catalogs like Glue.

Also, if we end up supporting other read APIs like PyIceberg's APIs, this will be more convenient.